### ISSUE TITLE
[Mega issue] - Accounts Revamp testing feedback

### ISSUE THREAD

**Issue by rkapka** [prysmaticlabs/prysm#6715]
# Information 

- [ ] **THIS SHOULD BE CHECKED ONLY AFTER ALL ISSUES HAVE BEEN CHECKED AND RETESTED**

Check an issue if it has been fixed and merged into `master`. Add the PR number to the issue.
Cross out an issue if it has been retested after fixing.
Check and cross out an issue if it will not be fixed. Please specify the reason for not fixing it (e.g. "not a bug because (...)", "not important enough to fix").

# Issues

## General

- [x] ~~A lot of commands display this prompt for the `--passwords-dir` flag:~~

~~>Directory where passwords will be stored~~

~~This wording doesn't make much sense for commands other than those that create password files (`wallet create` and `accounts import`).~~

- [x] ~~Changing the path for the `--wallet-dir` flag should update the default path for the `--passwords-dir` flag dynamically. It is annoying that I have to make the same changes again.~~

~~As an example, changing the wallet's directory to `/home/foo/bar/.prysm-wallet-v2` should default the password's directory to `/home/foo/bar/.prysm-wallet-v2-passwords`, not `/home/radek/.eth2validators/.prysm-wallet-v2-passwords`.~~

Won't fix in the interest of time and other priorities. Also, given this will be added to the wallet configuration, it might only be one time event to type it in.
- [x] ~~Many flag descriptions inform about the default directory. I think this is unnecessary because the default directory will be displayed if the flag is not provided. Having this information requires updating it whenever the default directory changes. Currently the default shown in flag descriptions does not contain the last segment of the path.~~
- [x] ~~The command names `wallet` and `accounts` are a bit inconsistent. The former is singular while the latter is plural.~~ (we typically manage one wallet, but manage multiple accounts. Given we have used the phrasing `accounts` in v1, it makes sense to keep it the same)

- [x] ~~Whenever I am prompted for any input and the input is longer than the available terminal width, typing every additional character results in a copy of the original line being added to the terminal. A similar thing happens while pasting to the terminal.~~

### Overall

- [x] I am not sure why the wallet's directory is hidden by default. I got confused when I couldn't find it after running the command for the first time. What adds to the confusion is that it will be visible after changing it to some other directory which does not start with a `.`. #6736

*Retest comment*: `.eth2validators` still begins with a `.`.

- [x] ~~The `edit-config` command should include the `--passwords-dir` flag because I got asked for this directory.~~ #6736
- [x] ~~The phrase "Enter the wallet recovery seed phrase you would like to recover" sounds to me as if I want to recover the seed phrase. It is used in the `recover` command.~~ #6725
- [x] It would be nice if the wallet's password path could be saved as part of the wallet's metadata. Typing this path over and over is super annoying.
- [x] ~~Run the `create` command and provide a directory already containing a wallet when prompted for the wallet directory. The error message is:~~

~~> FATAL accounts-v2: Could not create a wallet: could not check if wallet directory exists: you already have a wallet at the specified path. You can edit your wallet configuration by running ./prysm.sh validator wallet-v2 edit~~

~~The last word of the message should be `edit-config` instead of `edit`.~~ #6725 

- [x] ~~I don't understand how to use the `--password-flle` flag based on the help available in the console.~~

- [x] There is an info displayed after successfully creating a wallet:

> INFO accounts-v2: Successfully created wallet with on-disk keymanager configuration. Make a new validator account with ./prysm.sh validator accounts-v2 create wallet-path=/home/radek/eth2validators/pf/.prysm-wallet-v2/direct

It should be `--wallet-dir` instead of `wallet-path`. #6736 

*Retest comment*: The flag name displayed is still `wallet-path`.

- [x] ~~The command `create --help` lists a `--password-file` flag. This flag doesn't make sense for wallets.~~ #6736 

- [x] ~~Follow these steps:~~
    ~~a. Create an HD wallet~~
    ~~b. Create a non-HD wallet~~
    ~~c. Create one ore more accounts for the non-HD wallet~~
    ~~d. Export all accounts from the non-HD wallet~~
    ~~e. Import the exported accounts into the path of the HD wallet~~
    ~~f. Create a new account for the imported non-HD wallet~~

> ~~FATAL accounts-v2: Could not create new account: could not open wallet: could not read keymanager kind for wallet: wanted 1 directory in wallet dir, received 2~~

~~It should not be possible to mess up your wallets like this. Trying to perform step `e` should result in an error.~~ #6725 

### Non-HD

- [x] ~~The `.pass` files contain the account's password in plain-text. **any proposed solution @rkapka ?**~~

### HD

- [x] ~~The following prompt is displayed when creating a new wallet:~~

~~> Confirm you have written down the recovery words somewhere safe (offline)? [y/N]~~

~~Typing anything other than `y` results in the same behaviour, so the `N` option does not make much sense. Furthermore, typing `Y` does not confirm although it should. Maybe it would be better to change `[y/N]` to `[y/Y]`?~~ #6725

- [x] ~~When recovering the wallet I typed this pass phrase:~~

> ~~1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4~~

~~The message:~~

> ~~FATAL accounts-v2: Could not recover wallet: could not initialize new wallet seed file: could not convert mnemonic to wallet seed: word \`1\` not found in reverse map~~

~~Then I typed `critic` 24 times and got this:~~ 

> ~~FATAL accounts-v2: Could not recover wallet: could not initialize new wallet seed file: could not convert mnemonic to wallet seed: Checksum incorrect~~

~~In my opinion these messages are too technical. In both scenarios a unified, more user-friendly message should be displayed.~~ #6725 

### Remote

- [x] ~~Whenever I do not provide at least one flag out of the 4 gRPC configuration flags during `create` or `edit-config`, I am prompted for all 4 configuration values and I have to pass them again.~~ #6736

### Overall

- [x] ~~The description of the `create --help` command says~~ 

>~~If no account exists at the wallet path, creates a new wallet for a user~~

~~This doesn't feel right to me. Shouldn't it be `no wallet exists` instead of `no account exists`?~~ #6725

- [x] ~~panic here:~~
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x70 pc=0xfd1a0e]

goroutine 98 [running]:
github.com/manifoldco/promptui.(*Prompt).Run.func2(0x24c0798, 0x0, 0x0, 0x0, 0xd, 0xc000238590, 0x24c0798, 0x0, 0xc000238590, 0x162a1b0)
	external/com_github_manifoldco_promptui/prompt.go:166 +0x3de
github.com/chzyer/readline.(*DumpListener).OnChange(0xc000010560, 0x24c0798, 0x0, 0x0, 0x0, 0xd, 0x0, 0x0, 0x0, 0x0, ...)
	external/com_github_chzyer_readline/operation.go:516 +0x63
github.com/chzyer/readline.(*Operation).ioloop(0xc0007c4540)
	external/com_github_chzyer_readline/operation.go:339 +0x44d
created by github.com/chzyer/readline.NewOperation
	external/com_github_chzyer_readline/operation.go:88 +0x2b1
```
~~To recreate this error, repeatedly create a new account in the same directory. After a couple of tries you should reach this error after typing the account's password. I believe this error occurs when the account's name is duplicated.~~

 (Ignore for now, upstream issue https://github.com/manifoldco/promptui/issues/158)

- [x] ~~One step of the `export` command asks you:~~

> ~~Enter a file location to write the exported wallet to: home/radek/.eth2validators~~

~~I think it should say `exported accounts` instead of `exported wallet`. It would also be nice if the default path was dynamically adjusted to reflect the path from the `--wallet-dir` flag.~~

- [x] ~~The account list in the `export` command should be sorted alphabetically to make finding a specific account easier. The `All accounts` option should be listed first, not last. If I had a 1000 accounts, I would need to press the down arrow a lot to export all of them. Of course I could just use `--accounts all` instead, but I still think it would be better to have this option at the beginning. I assume most users will want to export all accounts and the option will be then immediately visible.~~ #6749 

- [x] ~~The exported `backup.zip` file gets overwritten without any confirmation.~~ #6749 
- [x] ~~When the `export` command displays the list of accounts to export, follow these steps:~~
    ~~- press Up~~
    ~~- press Enter~~
    ~~- type `blah`~~
    ~~- press Enter~~

~~Result:~~

> ~~FATAL accounts-v2: Could not export accounts: could not log out exported accounts: could not get public key: could not get keystore: could not read keystore file: could not read kkk: open /home/radek/eth2validators/non-hd/.prysm-wallet-v2/direct/kkk/keystore.json: no such file or directory~~

~~The account directory contains an empty directory called `blah` and it is no longer possible to export from this directory. The following is displayed:~~

> ~~FATAL accounts-v2: Could not export accounts: could not initialize keymanager: could not initialize direct keymanager: could not initialize keys cache: could not read password for account kkk: could not read /home/radek/eth2validators/non-hd-passwords/.prysm-wallet-v2-passwords/kkk.pass: open /home/radek/eth2validators/non-hd-passwords/.prysm-wallet-v2-passwords/kkk.pass: no such file or directory~~

- [x] ~~How can I export more that one account, but not all of them? Is there a way to choose multiple accounts from the list? If it is possible only via the `--accounts` flag, then how should the value be formatted? What is the separator? The help for this flag is not descriptive enough.~~ ( #6749 )
- [x] ~~Why does the `export` command contain a `--passwords-dir` flag when the passwords directory is not exported? Maybe it should be?~~ (Nishant: I dont think we should be exporting passwords in batch, imo just exporting the wallets should be ok. Also
we are not 100% sure on the status of plaintext passwords.)
- [x] ~~The description of import's `--backup-dir` is incorrect because it describes exporting.~~
- [x] ~~During importing the accounts get overwritten without any confirmation. This can lead to loss of accounts.~~ (#6749)
- [x] ~~The import is not an atomic operation and it probably should be. My accounts got imported when I cancelled the command before typing any passwords, and additionally one password file got created when I cancelled the command before typing the second password. Although it's technically OK to import the accounts without waiting for passwords (because the password will be required later to use the account anyway), it feels a bit strange that something happened even though I quit in the middle.~~
(Nishant: Not super important for launch, so we can ignore it but would be good to make the operation atomic in the future.) 

- [x] ~~I created a password file with password `fr%7tgxM` and passed it to the `--password-file` flag.~~ 

> ~~FATAL accounts-v2: Could not create new account: could not input new account password: password did not pass validation: password must only contain alphanumeric characters, punctuation, or symbols~~

~~I changed the password to a simpler one (`abc`) but the effect was the same.~~ #6725 

### HD

- [x] Providing a wrong password on account creation displays this: 

> FATAL accounts-v2: Could not create new account: could not initialize keymanager: could not initialize derived keymanager: could not decrypt seed configuration with password: invalid checksum

The message should just say that the password is incorrect. #6725 

*Retest comment*: Not fixed.

- [x] ~~The command `create --help` lists a `--skip-mnemonic-confirm` flag, but the mnemonic is a part of HD wallet creation, not account creation. Providing this flag when creating an HD wallet results in an unknown flag error.~~ #6725 
- [x] ~~There is a small difference in formatting between `list` for a non-HD wallet and an HD wallet.~~

~~This is Non-HD formatting:~~

> ~~ideally-composed-chipmunk~~
~~[public key] 0xa5a1c7904c7a49a0930cb5494e340ae08c0b53e129d8555a3147ea8df10807ba98525ee0ba5ba487d99d6b579f057a54~~
~~[created at] 1 day ago~~

~~This is HD formatting:~~

> ~~positively-relative-skunk | 1 hour ago~~
~~[withdrawal public key] 0xb0d4aff8a4557047858ee0d95115e453847680e3fc1d5f348e1814dafa4a666d776247e4e3d34c6876a3abda0a853f8a~~
~~[derivation path] m/12381/3600/0/0~~
~~[validating public key] 0x98585b12562cc6300658b9a1c3491b617e806d3eec672a2ba321658cacac0ad82ec184721651a53d987cefa115cb676a~~
~~[derivation path] m/12381/3600/0/0/0~~

~~The Non-HD one has the creation date on a separate `[created at]` line while the HD one has the creation date in the first line after a separator.~~ #6725 

### Non-HD

- [x] ~~`bazel run //validator accounts-v2 import` fails on a new dir~~
```Enter the directory where your keystores to import are located
/Users/shayzluf/Library/Eth2Validators/
Importing accounts: 
Importing accounts, this may take a while...

Imported 0 validator accounts
```

**Comment by raphpa** [comment]
Running `docker run --rm -ti -v /data/docker/medala:/data gcr.io/prysmaticlabs/prysm/validator wallet-v2 create` and trying to interactively creating a wallet:
When asking for a wallet directory `backspace` does not do anything and pressing the `delete` key crashes the process with 
```
FATAL accounts-v2: Could not create a wallet: could not check if wallet directory exists: could not parse wallet directory: could not determine directory: no input received, closing
```

**Comment by rkapka** [comment]
> I am not sure why the wallet's directory is hidden by default. I got confused when I couldn't find it after running the command for the first time. What adds to the confusion is that it will be visible after changing it to some other directory which does not start with a `.`.

One more thing I would like to add here. The directory of the wallet is needed to interact with that wallet's accounts. If the directory is hidden it will be difficult to find it.

**Comment by raphpa** [comment]
When you are manually specifying a path to the wallet directory interactively it always adds `.prysm-wallet-v2` to the given path. So you are actually have to specify the wallets' parent directory. I do not think this is the expected behaviour

**Comment by rauljordan** [comment]
Please re-check for closure @rkapka

**Comment by PandaWhisperer** [comment]
Not sure if this is the right issue for this, but I've just created some validator keys using the `eth2.0-deposit-cli`, and I am having trouble importing them into Prysm.

I'm using @stefa2k's [prysm-docker-compose](https://github.com/stefa2k/prysm-docker-compose) setup, but FWIW I'm essentially running the following command:

    accounts-v2 import --keys-dir="/launchpad/validator_keys" --wallet-dir="/data/wallets" --wallet-password-file="/data/passwords/wallet-password"

Of course I'm running this with docker, with the `/launchpad` mapped to `./launchpad` and the `/data` directory mapped to `./validator`. This is the error I'm getting:

    FATAL accounts-v2: Could not import accounts: could not initialize wallet: could not create new wallet: could not get password: password did not pass validation: password must have more than 8 characters, at least 1 special character, and 1 number

It appears that my password (which contains more than 8 character, 1 special character, **but NO number**) did not pass validation. This is strange, because the `eth2.0-deposit-cli` accepted my password as valid. It appears that the `account-vs` command is using a different password validation routine than the deposit CLI.

I've already made the deposits for the keys I've generated, and there appears to be no way of changing the password of the generated keys using the deposit CLI.

**Comment by stefa2k** [comment]
@PandaWhisperer actually pdc shouldn't ask a password if you [configure it properly](https://github.com/stefa2k/prysm-docker-compose/blob/master/validator/passwords/wallet-password) properly as described [here](https://github.com/stefa2k/prysm-docker-compose#validator-accounts-with-launchpad). If this is not working please ping me or create an issue at [pdc](https://github.com/stefa2k/prysm-docker-compose/issues) with a short information about it. Thanks!

**Comment by rkapka** [comment]
1. I am not sure why the wallet's directory is hidden by default. I got confused when I couldn't find it after running the command for the first time. What adds to the confusion is that it will be visible after changing it to some other directory which does not start with a `.`. #6736
    > `.eth2validators` still begins with a `.`.

2. There is an info displayed after successfully creating a wallet:
INFO accounts-v2: Successfully created wallet with on-disk keymanager configuration. Make a new validator account with ./prysm.sh validator accounts-v2 create wallet-path=/home/radek/eth2validators/pf/.prysm-wallet-v2/direct
It should be --wallet-dir instead of wallet-path.
    > The flag name displayed is still `wallet-path` instead of `--wallet-dir`.

3. Providing a wrong password on account creation displays this:
FATAL accounts-v2: Could not create new account: could not initialize keymanager: could not initialize derived keymanager: could not decrypt seed configuration with password: invalid checksum
The message should just say that the password is incorrect.
    > Not fixed.
